Source/KSCrash/Source/KSCrash/Recording/BSG_KSCrashReport.c

```diff
@@ -1635,3 +1617,59 @@ void bsg_kscrashreport_logCrash(const BSG_KSCrash_Context *const crashContext) {
     bsg_kscrw_i_logCrashType(crash);
     bsg_kscrw_i_logCrashThreadBacktrace(&crashContext->crash);
 }
+
+size_t thread_json_size = sizeof(char) * 32 * 1024;
+char *thread_json_data;
+
+int bsg_kscrw_i_collectJsonData(const char *const data, const size_t length, void *const userData) {
+    size_t new_size = strlen(thread_json_data) + length;
+    char tmp[length];
+    memcpy(tmp, data, length);
+    tmp[length] = '\0';
+
+    if (new_size > thread_json_size) {
+        thread_json_size *= 2;
+        thread_json_data = realloc(thread_json_data, thread_json_size);
+    }
+    strcat(thread_json_data, tmp);
+    return BSG_KSJSON_OK;
+}
+
+char *bsg_kscrw_i_captureThreadTrace(const BSG_KSCrash_Context *crashContext) {
+    if (thread_json_data == NULL) {
+        thread_json_data = malloc(thread_json_size);
+    }
+    memset(thread_json_data, 0, strlen(thread_json_data));
+    thread_json_data[0] = '\0';
```

Comment:
Can you move this to `bsg_kscrw_i_collectJsonData` so `thread_json_data` is managed in one place?

Reply:
Extracted to `bsg_kscrw_i_resetThreadTraceData` as suggested by in-person discussion.